baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StreamFlatMapOptional.java

```diff
@@ -30,50 +32,146 @@
+            MatchType.AT_LEAST_ONE,
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Optional.class.getCanonicalName())
+                    .named("stream")
+                    .withNoParameters());
 
-    private static final Matcher<ExpressionTree> OPTIONAL_STREAM = MethodMatchers.instanceMethod()
-            .onDescendantOf(Optional.class.getName())
-            .named("stream")
-            .withNoParameters();
+    private static final Matcher<ExpressionTree> STREAM_FILTER_IS_PRESENT = Matchers.methodInvocation(
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Stream.class.getCanonicalName())
+                    .named("filter")
+                    .withParameters(Predicate.class.getName()),
+            // Any of the three MatchTypes are reasonable in this case, given a single arg
+            MatchType.AT_LEAST_ONE,
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Optional.class.getCanonicalName())
+                    .named("isPresent")
+                    .withNoParameters());
 
-    private static final Matcher<ExpressionTree> STREAM_FLATMAP_OPTIONAL_STREAM = Matchers.methodInvocation(
-            STREAM_FLAT_MAP,
+    private static final Matcher<ExpressionTree> STREAM_MAP_GET = Matchers.methodInvocation(
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Stream.class.getCanonicalName())
+                    .named("map")
+                    .withParameters(Function.class.getName()),
             // Any of the three MatchTypes are reasonable in this case, given a single arg
-            MatchType.LAST,
-            OPTIONAL_STREAM);
+            MatchType.AT_LEAST_ONE,
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Optional.class.getCanonicalName())
+                    .namedAnyOf("get", "orElseThrow")
+                    .withNoParameters());
 
     @Override
     public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
         if (STREAM_FLATMAP_OPTIONAL_STREAM.matches(tree, state)) {
-            ExpressionTree receiver = ASTHelpers.getReceiver(tree.getMethodSelect());
-            if (receiver != null) {
-                SuggestedFix.Builder fix = SuggestedFix.builder();
-                String optionalType = SuggestedFixes.qualifyType(state, fix, Optional.class.getCanonicalName());
-                String replacement = ".filter(" + optionalType + "::isPresent).map(" + optionalType + "::get)";
-                return buildDescription(tree)
-                        .addFix(fix.replace(state.getEndPosition(receiver), state.getEndPosition(tree), replacement)
-                                .build())
-                        .build();
+            ExpressionTree methodSelect = tree.getMethodSelect();
+            ExpressionTree receiver = ASTHelpers.getReceiver(methodSelect);
```

Comment:
`ASTHelpers.getReceiver` follows `MethodInvocationTree.getMethodSelect()`, we can pass `tree` directly for the same result
```suggestion
            ExpressionTree receiver = ASTHelpers.getReceiver(tree);
```